Python/iSense.py

```diff
@@ -0,0 +1,310 @@
+        self.__created_at = dictionary['createdAt']
+        self.__owner_name = dictionary['ownerName']
+        self.__owner_url = dictionary['ownerUrl']
+        self.__data_set_count = dictionary['dataSetCount']
+        self.__data_set_ids = dictionary['dataSetIDs']
+        self.__field_count = dictionary['fieldCount']
+
+        # array of data set objects
+        self.__data_sets = []
+        for data_set in dictionary['dataSets']:
+            data_set_object = DataSet(data_set)
+            self.__data_sets.append(data_set_object)
+
+        self.__fields = []
+        self.__field_ids = []
+        for field in dictionary['fields']:
+            field_object = Field(field['id'], field['name'], field['type'], 
+                                unit=field['unit'], restrictions=field['restrictions'])
+            self.__field_ids.append(field['id'])
+            self.__fields.append(field_object)
+
+    def refresh(self):
+        self.__init__(self.__id)
+
+    def getId(self):
+        return self.__id
+
+    def getDataSets(self):
+        return self.__data_sets
+
+    def getDataSetById(self, data_id):
+        for d in self.__data_sets:
+            if d.getName() == data_id:
+                return d 
+
+        # None of the data sets match
+        return None
+
+    def getDataSetByName(self, name):
+        for d in self.__data_sets:
+            if d.getName() == name:
+                return d
+
+        # None of the data sets match
+        return None
+
+    def getFields(self):
+        return self.__fields
+
+    def getFieldIds(self):
+        return self.__field_ids
+
+    # Other can be None, units if field is a number, or an array of text restrictions if the field is a text field
+    def addField(self, credentials, name, fieldType, other):
+        # Only the owner of the project can add fields
+        if credentials.isUser():
+            response = iSenseAPI.createField(credentials.getUsername(), credentials.getPassword(), self.__id, name, fieldType, other)
+            self.refresh()
+            return response
+
+        return INVALID_CREDENTIALS_MESSAGE
+
+
+    #   createDataSet takes a dictionary like this
+    #   {
+    #       "FIELD_ID" => [1,2,3,4,5], # This would be like a column in a spread sheet (the field would be the header)
+    #       "FIELD_ID_2" => ["blue","red",,,"green"] # This would be another column in a spread sheet. (the empty spots will show up as blank entries for those rows)
+    #   }
+
+    def createDataSet(self, title, data, credentials):
+        # post request too post data
+        if credentials.isUser():
+            result =  iSenseAPI.uploadDataSet(credentials.getUsername(), credentials.getPassword(), self.__id, title, data)
+        elif credentials.isKey():
+            result =  iSenseAPI.uploadDataSetWithKey(credentials.getContribKey(), credentials.getContribName(), self.id, title, data)
+        else:
+            result = INVALID_CREDENTIALS_MESSAGE
```

Comment:
See comment below.  Bad credentials (ie: can't log in) should be thrown here.  Creating an invalid credentials object should be thrown when you create the credentials object.  In general, don't return strings as error messages.